ShapesProject/Project1/TDynamicArray.h

```diff
@@ -0,0 +1,320 @@
+	}
+
+	//------------------------------------------------------------------------------------------
+
+	void PopBack()
+	{
+		delete DynamicArray[Size];
+		Size--;
+	}
+
+	//------------------------------------------------------------------------------------------
+
+	void Insert(const T& NewItem, const int Position)
+	{
+		CheckBounds();
+
+		Size++;
+
+		//Auxiliar variables
+		T AuxiliarA{ DynamicArray[Position] };
+		T AuxiliarB{ DynamicArray[Position] };
+
+		for (int i = Position; i < Size - 1; i++)
+		{
+			//moves the values +1 position
+			AuxiliarA = DynamicArray[i + 1];
+			DynamicArray[i + 1] = AuxiliarB;
+			AuxiliarB = AuxiliarA;
+		}
+		DynamicArray[Position] = NewItem;
+	}
+
+	//------------------------------------------------------------------------------------------
+
+	void Erase(const int Position)
+	{
+		CheckBounds();
+
+		for (int i = Position; i < Size - 1; i++)
+		{
+			DynamicArray[i] = DynamicArray[i + 1];
+		}
+
+		delete DynamicArray[Size];
+		Size--;
+	}
+
+	//------------------------------------------------------------------------------------------
+
+	void Fill(const T& Value)
+	{
+		for (int i = 0; i < Size; i++)
+		{
+			DynamicArray[i] = Value;
+		}
+	}
+
+	//------------------------------------------------------------------------------------------
+
+	bool Swap(TDynamicArray& OtherArray)
+	{
+		int OtherSize = sizeof(OtherArray) / sizeof(OtherArray[0]);
+
+		if (OtherSize == Size)
+		{
+			T* AuxiliarArray = new T[Size];
+			memcpy(AuxiliarArray, DynamicArray, Size);
+			memcpy(DynamicArray, OtherArray, Size);
+			memcpy(OtherArray, AuxiliarArray, Size);
+
+			delete[] OtherArray;
+			return true;
+		}
+		return false;
+	}
+
+	//------------------------------------------------------------------------------------------
+
+	void Append(const TDynamicArray& OtherArray)
+	{
+		int OtherArraySize = sizeof(OtherArray) / sizeof(T);
+		int NewSize = Size + OtherArraySize;
+		Resize(NewSize);
+
+		int OtherIndex = 0;
+
+		for (int i = Size; i < NewSize; i++)
+		{
+			DynamicArray[i] = OtherArray[OtherIndex];
+			OtherIndex++;
+		}
+
+		CheckBounds();
+	}
+
+	//------------------------------------------------------------------------------------------
+
+	void Resize(const int NewCapacity)
+	{
+		T* NewDynamicArray = new T[NewCapacity];
+
+		for (int i = 0; i < Size; i++)
+		{
+			NewDynamicArray[i] = DynamicArray[i];
+		}
+
+		delete[] DynamicArray;
+		DynamicArray = NewDynamicArray;
+
+		delete[] NewDynamicArray;
+
+	}
+
+	void CheckBounds()
+	{
+		if (Size >= Capacity)
+		{
+			Reserve();
+		}
+	}
+
+	bool AddUnique(const T& NewElement)
+	{
+		for (int i = 0; i < Size; i++)
+		{
+			if (DynamicArray[i] == NewElement)
+			{
+				return false;
+			}
+		}
+		PushBack(NewElement);
+		return true;
+	}
+
+	bool Contains(const T& InObject) const
+	{
+		for (T LoopObject: DynamicArray)
+		{
+			if (LoopObject == InObject)
+			{
+				return true;
+			}
+		}
+		return false;
+	}
+
+	int GetIndex(const T& InObject) const
```

Comment:
This name is not incorrect nonetheless it doesn't make it clear for the developer or user that you're gonna return the index of its first appearance in the array. This function is actually often called Find in other similar classes such as TArrays